pkg/secretlock/local/masterkey_secret.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package local
+
+import (
+	"crypto/cipher"
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"hash"
+	"io"
+
+	"github.com/google/tink/go/subtle/random"
+	"golang.org/x/crypto/hkdf"
+
+	"github.com/hyperledger/aries-framework-go/pkg/secretlock"
+)
+
+type masterLockHKDF struct {
+	h    func() hash.Hash
+	salt []byte
+	aead cipher.AEAD
+}
+
+// NewMasterLock is responsible for encrypting/decrypting a master key
+// for the passed in `passphrase`, hash function `h`, `salt`.
+// The size of a master key passed to Encrypt() must match `h()`.Size() since the key used for
+// encryption/decryption is expanded from passphrase using HKDF.
+// The salt is optional and can be set to nil.
+// This implementation must not be used directly in Aries framework. It should be passed in
+// as the second argument of NewService(masterKeyReader io.Reader, secLock secretlock.Service) of this local package.
+func NewMasterLock(passphrase string, h func() hash.Hash, salt []byte) (secretlock.Service, error) {
```

Comment:
Shouldn't the function name reflect that this is a HKDF or passphrase?

Reply:
technically this function is the constructor of `masterLockHKDF`

Reply:
master key lock is now in it's own sub package, so this function would be called as follows:
`hdkf.NewMasterLock()`